tests/parser.cpp

```diff
@@ -2479,20 +2468,20 @@ k:f { print(5e-9); }
       ~~~~~~~~
 )");
 
-  test_parse_failure("k:f { print(1e17); }", R"(
-stdin:1:7-17: ERROR: Exponent will overflow integer range: 17
-k:f { print(1e17); }
+  test_parse_failure("k:f { print(1e21); }", R"(
+stdin:1:7-17: ERROR: overflow error, maximum value is 18446744073709551615: 1e21
+k:f { print(1e21); }
       ~~~~~~~~~~
 )");
 
   test_parse_failure("k:f { print(12e4); }", R"(
-stdin:1:7-17: ERROR: Coefficient part of scientific literal must be in range (0,9), got: 12
+stdin:1:7-17: ERROR: coefficient part of scientific literal must be 1-9: 12e4
 k:f { print(12e4); }
       ~~~~~~~~~~
 )");
 
   test_parse_failure("k:f { print(1_1e100); }", R"(
-stdin:1:7-20: ERROR: Coefficient part of scientific literal must be in range (0,9), got: 11
+stdin:1:7-20: ERROR: coefficient part of scientific literal must be 1-9: 11e100
```

Comment:
we have quite a mix of capitalization at the beginning of ERROR sentences - even in this PR 😉

Reply:
I tend to anchor on always lower case for errors, but should capitalize log messages (and use complete sentences) when you are actually controlling the output. I'm used to this standard from Go (e.g. https://staticcheck.dev/docs/checks#ST1005).  This lets you chain errors effectively. Anyways, call me out if I'm doing something inconsistent from *that* angle (e.g. capitalizing the errors).